### PR TITLE
[SPG-823]: adding pipeline filters TF resource

### PR DESCRIPTION
**Pull request by RajBaviskar**
## Describe your changes

- adding pipeline filters TF resource

## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

File: internal/service/platform/pipeline_filters/resource_pipeline_filters.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+			"name": {
+				Description: "Name of the pipeline filters",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"type": {
+				Description:  "Type of pipeline filters. Currently supported types are {Connector, DelegateProfile, Delegate, PipelineSetup, PipelineExecution, Deployment, Audit, Template, EnvironmentGroup, FileStore, CCMRecommendation, Anomaly, Environment}",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Connector", "DelegateProfile", "Delegate", "PipelineSetup", "PipelineExecution", "Deployment", "Audit", "Template", "EnvironmentGroup", "FileStore", "CCMRecommendation", "Anomaly", "Environment"}, false),
```
Comment (rathodmeetsatish) [inline internal/service/platform/pipeline_filters/resource_pipeline_filters.go:40]: for pipeline filters i guess all these types wont be valid like connector or audit similarly in data source

File: internal/service/platform/pipeline_filters/resource_pipeline_filters.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"filter_type": {
+							Description:  "Corresponding Entity of the filters. Currently supported types are {Connector, DelegateProfile, Delegate, PipelineSetup, PipelineExecution, Deployment, Audit, Template, EnvironmentGroup, FileStore, CCMRecommendation, Anomaly, Environment}.",
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"Connector", "DelegateProfile", "Delegate", "PipelineSetup", "PipelineExecution", "Deployment", "Audit", "Template", "EnvironmentGroup", "FileStore", "CCMRecommendation", "Anomaly", "Environment"}, false),
```
Comment (rathodmeetsatish) [inline internal/service/platform/pipeline_filters/resource_pipeline_filters.go:63]: same as above

File: internal/service/platform/pipeline_filters/resource_pipeline_filters.go (hunk 1 above)
Comment (RajBaviskar) [reply internal/service/platform/pipeline_filters/resource_pipeline_filters.go]: updated

File: internal/service/platform/pipeline_filters/resource_pipeline_filters.go (hunk 2 above)
Comment (RajBaviskar) [reply internal/service/platform/pipeline_filters/resource_pipeline_filters.go:63]: updated

File: internal/service/platform/pipeline_filters/data_source_pipeline_filters.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+func DataSourcePipelineFilters() *schema.Resource {
+	resource := &schema.Resource{
+		Description: "Data source for retrieving a Harness Pipeline Filter.",
+
+		ReadContext: dataSourcePipelineFiltersRead,
+
+		Schema: map[string]*schema.Schema{
+			"identifier": {
+				Description: "Unique identifier of the resource",
```
Comment (rijajoo) [inline internal/service/platform/pipeline_filters/data_source_pipeline_filters.go:24]: Please add full stops (.) at the end of every description sentence.

File: internal/service/platform/pipeline_filters/data_source_pipeline_filters.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+				Computed:    true,
+			},
+			"type": {
+				Description:  "Type of filter. Currently supported types are {PipelineSetup, PipelineExecution, Deployment, Template, EnvironmentGroup, Environment}.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"PipelineSetup", "PipelineExecution", "Deployment", "Template", "EnvironmentGroup", "Environment"}, false),
+			},
+			"org_id": {
+				Description: "organization Identifier for the Entity",
```
Comment (rijajoo) [inline internal/service/platform/pipeline_filters/data_source_pipeline_filters.go:40]: Please check that sentences start with Capital case.

**Review by rijajoo — CHANGES_REQUESTED** [review]

File: internal/service/platform/pipeline_filters/data_source_pipeline_filters.go (hunk 4 above)
Comment (RajBaviskar) [reply internal/service/platform/pipeline_filters/data_source_pipeline_filters.go]: changed.

File: docs/resources/platform_pipeline_filters.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "harness_platform_pipeline_filters Resource - terraform-provider-harness"
+subcategory: "Next Gen"
+description: |-
+  Resource for creating a Harness Pipeline Filters.
+---
+
+# harness_platform_pipeline_filters (Resource)
+
+Resource for creating a Harness Pipeline Filters.
```
Comment (Clare-Wilkinson) [inline docs/resources/platform_pipeline_filters.md:11]: As above, please remove "a" from before Harness.

**Review by Clare-Wilkinson — COMMENTED** [review]
Thanks for sending - a few small things just :)

**Pull request merged**